java-frontend/src/main/java/org/sonar/java/model/LiteralUtils.java

```diff
@@ -171,7 +171,7 @@ private static String stripIndent(int indent, String s) {
   }
 
   private static String removeTrailingSpaces(String s) {
-    return s.replaceAll("[\t \f]*$", "");
+    return s.replaceAll("\\s+$", "");
```

Comment:
remove catastrophic backtracking risk ([java:S5852](https://next.sonarqube.com/sonarqube/coding_rules?open=java%3AS5852&rule_key=java%3AS5852))

Reply:
The fix should be `[\t \f]++$`